waku-bindings/src/node/mod.rs

```diff
@@ -126,56 +138,97 @@ impl WakuNodeHandle<Running> {
     }
 
     /// Closes the pubsub subscription to stop receiving messages matching a content filter. No more messages will be received from this pubsub topic
-    pub fn relay_unsubscribe(&self, pubsub_topic: &PubsubTopic) -> Result<()> {
-        relay::waku_relay_unsubscribe(&self.ctx, pubsub_topic)
+    pub async fn relay_unsubscribe(&self, pubsub_topic: &PubsubTopic) -> Result<()> {
+        relay::waku_relay_unsubscribe(&self.ctx, pubsub_topic).await
     }
 
-    pub fn filter_subscribe(
+    pub async fn filter_subscribe(
         &self,
         pubsub_topic: &PubsubTopic,
         content_topics: Vec<WakuContentTopic>,
     ) -> Result<()> {
-        filter::waku_filter_subscribe(&self.ctx, pubsub_topic, content_topics)
+        filter::waku_filter_subscribe(&self.ctx, pubsub_topic, content_topics).await
     }
 
-    pub fn filter_unsubscribe(
+    pub async fn filter_unsubscribe(
         &self,
         pubsub_topic: &PubsubTopic,
         content_topics: Vec<WakuContentTopic>,
     ) -> Result<()> {
-        filter::waku_filter_unsubscribe(&self.ctx, pubsub_topic, content_topics)
+        filter::waku_filter_unsubscribe(&self.ctx, pubsub_topic, content_topics).await
     }
 
-    pub fn filter_unsubscribe_all(&self) -> Result<()> {
-        filter::waku_filter_unsubscribe_all(&self.ctx)
+    pub async fn filter_unsubscribe_all(&self) -> Result<()> {
+        filter::waku_filter_unsubscribe_all(&self.ctx).await
     }
 
-    pub fn lightpush_publish_message(
+    pub async fn lightpush_publish_message(
         &self,
         message: &WakuMessage,
         pubsub_topic: &PubsubTopic,
     ) -> Result<MessageHash> {
-        lightpush::waku_lightpush_publish_message(&self.ctx, message, pubsub_topic)
+        lightpush::waku_lightpush_publish_message(&self.ctx, message, pubsub_topic).await
+    }
+
+    pub async fn store_query(
+        &self,
+        pubsub_topic: Option<PubsubTopic>,
+        content_topics: Vec<WakuContentTopic>,
+        peer_addr: &str,
+    ) -> Result<Vec<StoreWakuMessageResponse>> {
+        let one_day_in_secs = 60 * 60 * 24;
+        let time_start = (Duration::from_secs(Utc::now().timestamp() as u64)
+            - Duration::from_secs(one_day_in_secs))
+        .as_nanos() as usize;
+
+        let mut cursor: Option<MessageHash> = None;
+
+        let mut messages: Vec<StoreWakuMessageResponse> = Vec::new();
+
+        loop {
+            let response = store::waku_store_query(
+                &self.ctx,
+                "hard-coded-req-id".to_string(),
+                true, // include_data
+                pubsub_topic.clone(),
+                content_topics.clone(),
+                Some(time_start), // time_start
+                None,             // end_time
+                None,             // message_hashes
+                cursor,           // pagination_cursor
+                true,             // pagination_forward
+                Some(25),         // pagination_limit,
```

Comment:
Should probably be parameters as well.

Perhaps for all of the parameters of the query it makes sense to define a query object that can be built with an options pattern (https://medium.com/@omid.jn/options-pattern-in-rust-6425520b2b23)